subsys/debug/cpu_load/cpu_load.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <debug/cpu_load.h>
+#include <shell/shell.h>
+#ifdef DPPI_PRESENT
+#include <nrfx_dppi.h>
+#else
+#include <nrfx_ppi.h>
+#endif
+#include <helpers/nrfx_gppi.h>
+#include <nrfx_timer.h>
+#include <hal/nrf_rtc.h>
+#include <hal/nrf_clock.h>
+#include <hal/nrf_power.h>
+#include <debug/ppi_trace.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(cpu_load, CONFIG_CPU_LOAD_LOG_LEVEL);
+
+/* Convert event address to associated publish register */
+#define PUBLISH_ADDR(evt) (volatile u32_t *)(evt + 0x80)
+
+/* Define to please compiler when periodic logging is disabled. */
+#ifndef CONFIG_CPU_LOAD_LOG_INTERVAL
+#define CONFIG_CPU_LOAD_LOG_INTERVAL 0
+#endif
```

Comment:
It is not allowed to define macros that are in the Kconfig namespace, like CONFIG_CPU_LOAD_LOG_INTERVAL.

You can either have CONFIG_CPU_LOAD_LOG_INTERVAL not depend on PERIODIC, and have the default value be 0 when not PERIODIC.

Or instead of referencing CONFIG_CPU_LOAD_LOG_INTERVAL in code, you can reference CPU_LOAD_LOG_INTERVAL.

Reply:
> Or instead of referencing CONFIG_CPU_LOAD_LOG_INTERVAL in code, you can reference CPU_LOAD_LOG_INTERVAL.

You mean something like this?
```
#ifdef CONFIG_CPU_LOAD_LOG_INTERVAL 
#define CPU_LOAD_LOG_INTERVAL CONFIG_CPU_LOAD_LOG_INTERVAL 
#else 
#define CPU_LOAD_LOG_INTERVAL 0
#else 
```

Reply:
yes

Reply:
done